File: python/src/exampleA.py

class ShoppingBasket:

    def add(self, item, quantity):
        raise NotImplementedError()

    def get_quantity(self, item):
        raise NotImplementedError()

    def calculate_total(self):
        raise NotImplementedError()


class BasketItem:

    def __init__(self, name, price):
        raise NotImplementedError()

File: typescript/src/exampleA.ts

export class ShoppingBasket {
  add(item: BasketItem, quantity: number): void {
    throw new Error("Not implemented");
  }

  get_quantity(item: BasketItem): number {
    throw new Error("Not implemented");
  }

  calculate_total(): number {
    throw new Error("Not implemented");
  }
}

export class BasketItem {
  constructor(name: string, price: number) {
    throw new Error("Not implemented");
  }
}
